src/storage/src/sst/parquet.rs

```diff
@@ -157,30 +157,34 @@ impl<'a> ParquetReader<'a> {
 
         let adapter = ReadAdapter::new(store_schema.clone(), self.projected_schema.clone())?;
 
-        let pruned_row_groups = self.predicate.prune_row_groups(
-            store_schema.schema().clone(),
-            builder.metadata().row_groups(),
-        );
+        let pruned_row_groups = self
+            .predicate
+            .prune_row_groups(
+                store_schema.schema().clone(),
+                builder.metadata().row_groups(),
+            )
+            .into_iter()
+            .enumerate()
+            .flat_map(|(idx, valid)| if valid { Some(idx) } else { None })
```

Comment:
You should use filter_map instead of flat_map. flat_map is used to flatten an iterator